mwdb/resources/file.py

```diff
@@ -470,6 +470,124 @@ def post(self, identifier):
         return schema.dump({"token": download_token})
 
 
+@rate_limited_resource
+class FileDownloadResourceXored(Resource):
+    """copy of FileDownloadResource with file_obj.iterate_xored() in .get method"""
+    def get(self, identifier):
+        """
+        ---
+        summary: Download file
+        description: |
+            Returns file contents.
+
+            Optionally accepts file download token to get
+            the file via direct link (without Authorization header)
+        security:
+            - bearerAuth: []
+        tags:
+            - file
+        parameters:
+            - in: path
+              name: identifier
+              schema:
+                type: string
+              description: File identifier (SHA256/SHA512/SHA1/MD5)
+            - in: query
+              name: token
+              schema:
+                type: string
+              description: |
+                File download token for direct link purpose
+              required: false
+        responses:
+            200:
+                description: File contents
+                content:
+                  application/octet-stream:
+                    schema:
+                      type: string
+                      format: binary
+            403:
+                description: |
+                    When file download token is no longer valid
+                    or was generated for different object
+            404:
+                description: |
+                    When file doesn't exist, object is not a file
+                    or user doesn't have access to this object.
+            503:
+                description: |
+                    Request canceled due to database statement timeout.
+        """
+        access_token = request.args.get("token")
+
+        if access_token:
+            file_obj = File.get_by_download_token(access_token)
+            if not file_obj:
+                raise Forbidden("Download token expired, please re-request download.")
+            if not (
+                file_obj.sha1 == identifier
+                or file_obj.sha256 == identifier
+                or file_obj.sha512 == identifier
+                or file_obj.md5 == identifier
+            ):
+                raise Forbidden(
+                    "Download token doesn't apply to the chosen object. "
+                    "Please re-request download."
+                )
+        else:
+            if not g.auth_user:
+                raise Unauthorized("Not authenticated.")
+            file_obj = File.access(identifier)
+            if file_obj is None:
+                raise NotFound("Object not found")
+
+        return Response(
+            file_obj.iterate_xored(),
+            content_type="application/octet-stream",
+            headers={"Content-disposition": f"attachment; filename={file_obj.sha256}"},
+        )
+
+    @requires_authorization
+    def post(self, identifier):
```

Comment:
As far as I can see, this is exactly the same code as FileDownloadResource.post. Can you remove the dupllication? It's probably as easy as replacing this whole definition with

```
post = FileDownloadResource.post
```

But do test that. Otherwise maybe you can call that  one from this function.